ford/sourceform.py

```diff
@@ -923,6 +923,40 @@ def __init__(
                 else:
                     self.print_error(line, "Unexpected USE statement")
             else:
+                def get_call_chain(match) -> dict:
+                    bracket = 0
+                    spacer = False
+                    call_chain = []
+                    tmp = ''
+                    for c in line[match.start(1)-1::-1]:
+                        if bracket < 0:
+                            break
+                        if c == ')':
+                            bracket += 1
+                        elif c == '(':
+                            bracket -= 1
+                        elif bracket == 0:
+                            if c.isalnum() or c == '_':
+                                if spacer:
+                                    if len(tmp) == 0:
+                                        spacer = False
+                                    else:
+                                        break
+                                tmp = c + tmp
+                            elif c == '%':
+                                if(len(tmp) != 0):
+                                    call_chain.append(tmp.lower())
+                                    tmp = ''
+                            elif c == ' ':
+                                spacer = True
+                                continue
+                            else:
+                                break
+                    if(len(tmp) != 0):
+                        call_chain.append(tmp.lower())
+                    call_chain.reverse()
+                    return {"name": match.group(1).lower(), "call_chain": call_chain}
```

Comment:
After pulling out this function, it might be worth using a `dataclass` -- faster access and better type hinting